do_kafka_test.sh

```diff
@@ -0,0 +1,153 @@
+# !/bin/bash
+
+# This script is used to test the S3D application.
+# It starts Kafka, S3D, and pushes test data to Kafka.
+# It then kills S3D and Kafka.
+
+# Make sure to set the following environment variables:
+# - DOCKER_HOST_IP
+# - DEPOSIT_GROUP
+# - DEPOSIT_KEY_NAME
+# - DEPOSIT_BUCKET_NAME
+# - AWS_REGION
+# - DEPOSIT_TOPIC
+# - HEADER_X_API_KEY
+
+setup() {
+    # build if target folder does not exist
+    if [ ! -d "target" ]; then
+        echo "Building."
+        mvn clean package assembly:single
+    else
+        echo "Target folder exists. Skipping build."
+    fi
+
+    # if any of the required environment variables are not set, exit
+    if [ -z "$DOCKER_HOST_IP" ]; then
+        echo "DOCKER_HOST_IP is not set. Exiting."
+        exit 1
+    elif [ -z "$DEPOSIT_GROUP" ]; then
+        echo "DEPOSIT_GROUP is not set. Exiting."
+        exit 1
+    elif [ -z "$DEPOSIT_KEY_NAME" ]; then
+        echo "DEPOSIT_KEY_NAME is not set. Exiting."
+        exit 1
+    elif [ -z "$DEPOSIT_BUCKET_NAME" ]; then
+        echo "DEPOSIT_BUCKET_NAME is not set. Exiting."
+        exit 1
+    elif [ -z "$AWS_REGION" ]; then
+        echo "AWS_REGION is not set. Exiting."
+        exit 1
+    elif [ -z "$DEPOSIT_TOPIC" ]; then
+        echo "DEPOSIT_TOPIC is not set. Exiting."
+        exit 1
+    elif [ -z "$HEADER_X_API_KEY" ]; then
+        echo "HEADER_X_API_KEY is not set. Exiting."
+        exit 1
+    fi
+
+    useEndpoint=false
+
+    # if API_ENDPOINT is set, set useEndpoint to true
+    if [ ! -z "$API_ENDPOINT" ]; then
+        useEndpoint=true
+        echo "API_ENDPOINT is set. Using API_ENDPOINT."
+    fi
+
+    projectFolder=$(pwd | awk -F/ '{print $NF}')
+    EXPECTED_KAFKA_CONTAINER_NAME="$projectFolder-kafka-1"
+    echo "EXPECTED_KAFKA_CONTAINER_NAME: $EXPECTED_KAFKA_CONTAINER_NAME"
+
+    ./start_kafka.sh
+}
+
+waitForKafkaToCreateTopics() {
+    while true; do
+        if [ $(docker ps | grep $EXPECTED_KAFKA_CONTAINER_NAME | wc -l) == "0" ]; then
+            echo "Kafka container '$EXPECTED_KAFKA_CONTAINER_NAME' is not running. Exiting."
+            ./stop_kafka.sh
+            exit 1
+        fi
+
+        ltopics=$(docker exec -it $EXPECTED_KAFKA_CONTAINER_NAME /opt/kafka/bin/kafka-topics.sh --list --zookeeper 172.17.0.1)
```

Comment:
Where's this zookeeper ip coming from? Are we guaranteed that ip?

Reply:
I don't think so, good catch. It looks like this is supposed to be the IP address of the host machine running zookeeper, so I've changed this to use the value of the DOCKER_HOST_IP environment variable.

I verified the functionality, but I believe it was also working with `172.17.0.1`, interestingly.